From b0f492caf6afccf75e7431f7cd357b427decdcde Mon Sep 17 00:00:00 2001
From: Stanley.Miao <stanley.miao@windriver.com>
Date: Mon, 14 Mar 2011 16:39:52 +0800
Subject: [PATCH 006/108] OMAP3: SDRC: Add timing data for Hynix H8KDS0UN0MER-4EM

commit 62bad1c713471d719a9b53f5317aaf90ffad43de
in git://arago-project.org/git/projects/linux-omap3.git

The processor AM/DM3730 are using the PoP memory Hynix (H8KDS0UN0MER-4EM).

Signed-off-by: Ranjith Lohithakshan <ranjithl@ti.com>
Integrated-by: Stanley.Miao <stanley.miao@windriver.com>
---
 arch/arm/mach-omap2/board-omap3evm.c               |    6 +++-
 arch/arm/mach-omap2/sdram-hynix-h8kds0un0mer-4em.h |   37 ++++++++++++++++++++
 2 files changed, 42 insertions(+), 1 deletions(-)
 create mode 100644 arch/arm/mach-omap2/sdram-hynix-h8kds0un0mer-4em.h

diff --git a/arch/arm/mach-omap2/board-omap3evm.c b/arch/arm/mach-omap2/board-omap3evm.c
index bbdc373..8a365e3 100644
--- a/arch/arm/mach-omap2/board-omap3evm.c
+++ b/arch/arm/mach-omap2/board-omap3evm.c
@@ -48,6 +48,7 @@
 
 #include "mux.h"
 #include "sdram-micron-mt46h32m32lf-6.h"
+#include "sdram-hynix-h8kds0un0mer-4em.h"
 #include "hsmmc.h"
 
 #define OMAP3_EVM_TS_GPIO	175
@@ -813,7 +814,10 @@ static void __init omap3_evm_init_irq(void)
 {
 	omap_board_config = omap3_evm_config;
 	omap_board_config_size = ARRAY_SIZE(omap3_evm_config);
-	omap2_init_common_hw(mt46h32m32lf6_sdrc_params, NULL);
+	if (cpu_is_omap3630())
+		omap2_init_common_hw(h8kds0un0mer4em_sdrc_params, NULL);
+	else
+		omap2_init_common_hw(mt46h32m32lf6_sdrc_params, NULL);
 	omap_init_irq();
 	omap_gpio_init();
 }
diff --git a/arch/arm/mach-omap2/sdram-hynix-h8kds0un0mer-4em.h b/arch/arm/mach-omap2/sdram-hynix-h8kds0un0mer-4em.h
new file mode 100644
index 0000000..c147586
--- /dev/null
+++ b/arch/arm/mach-omap2/sdram-hynix-h8kds0un0mer-4em.h
@@ -0,0 +1,37 @@
+/*
+ * SDRC register values for the Hynix H8KDS0UN0MER-4EM
+ *
+ * Copyright (C) 2010 Texas Instruments, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#ifndef __ARCH_ARM_MACH_OMAP2_SDRAM_HYNIX_H8KDS0UN0MER4EM
+#define __ARCH_ARM_MACH_OMAP2_SDRAM_HYNIX_H8KDS0UN0MER4EM
+
+#include <plat/sdrc.h>
+
+/* Hynix H8KDS0UN0MER-4EM */
+static struct omap_sdrc_params h8kds0un0mer4em_sdrc_params[] = {
+	[0] = {
+		.rate        = 200000000,
+		.actim_ctrla = 0x92e1c4c6,
+		.actim_ctrlb = 0x0002111c,
+		.rfr_ctrl    = 0x0005e601,
+		.mr          = 0x00000032,
+	},
+	[1] = {
+		.rate        = 100000000,
+		.actim_ctrla = 0x49912283,
+		.actim_ctrlb = 0x0002110e,
+		.rfr_ctrl    = 0x0002da01,
+		.mr          = 0x00000032,
+    },
+	[2] = {
+		.rate        = 0
+	},
+};
+
+#endif
-- 
1.7.0.4

